### PR TITLE
Add properties that turn off the import of all capabiltiies so that shared projects can turn them off

### LINKED ISSUE #428: Visual Studio crash opening Roslyn in wpt branch
Ported from https://github.com/dotnet/roslyn/issues/13517#event-775358343

This is in WPT branch.

ArgumentNullException:
Value cannot be null.
Parameter name: path1
```
Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.ErrorReporting.FatalError.Report(System.Exception exception, System.Action<System.Exception> handler)	Unknown
 	Microsoft.VisualStudio.LanguageServices.VisualBasic.dll!Microsoft.VisualStudio.LanguageServices.VisualBasic.ProjectSystemShim.VisualBasicProject.SetCompilerOptions(ref Microsoft.VisualStudio.LanguageServices.VisualBasic.ProjectSystemShim.Interop.VBCompilerOptions pCompilerOptions)	Unknown
 	[Native to Managed Transition]	
 	mscorlib.dll!System.IO.Path.Combine(string path1, string path2)	Unknown
 	Microsoft.VisualStudio.LanguageServices.dll!Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Legacy.AbstractLegacyProject.TryGetOutputPathFromHierarchy(Microsoft.VisualStudio.Shell.Interop.IVsHierarchy hierarchy, string containingDirectoryPathOpt, out string binOutputPath)	Unknown
>	Microsoft.VisualStudio.LanguageServices.dll!Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Legacy.AbstractLegacyProject.TryGetOutputPathFromHierarchy(out string binOutputPath)	Unknown
 	Microsoft.VisualStudio.LanguageServices.dll!Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.AbstractProject.SetOutputPathAndRelatedData(string objOutputPath, bool hasSameBinAndObjOutputPaths)	Unknown
 	Microsoft.VisualStudio.LanguageServices.VisualBasic.dll!Microsoft.VisualStudio.LanguageServices.VisualBasic.ProjectSystemShim.VisualBasicProject.UpdateOptions()	Unknown
 	Microsoft.VisualStudio.LanguageServices.VisualBasic.dll!Microsoft.VisualStudio.LanguageServices.VisualBasic.ProjectSystemShim.VisualBasicProject.SetCompilerOptions(ref Microsoft.VisualStudio.LanguageServices.VisualBasic.ProjectSystemShim.Interop.VBCompilerOptions pCompilerOptions)	Unknown
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	Microsoft.VisualStudio.ProjectSystem.Managed.VS.dll!Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices.AbstractLanguageServiceHost.InitializeAsync.AnonymousMethod__50_0()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.InvokeMoveNext(object stateMachine)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.Run()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.YieldAwaitable.YieldAwaiter.RunAction(object state)	Unknown
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTaskFactory.SingleExecuteProtector.TryExecute()	Unknown
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTaskFactory.SingleExecuteProtector..cctor.AnonymousMethod__20_0(object state)	Unknown
 	WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate callback, object args, int numArgs)	Unknown
 	WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.TryCatchWhen(object source, System.Delegate callback, object args, int numArgs, System.Delegate catchHandler)	Unknown
 	WindowsBase.dll!System.Windows.Threading.DispatcherOperation.InvokeImpl()	Unknown
 	WindowsBase.dll!System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(object state)	Unknown
 	WindowsBase.dll!MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(object obj)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state)	Unknown
 	WindowsBase.dll!MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext executionContext, System.Threading.ContextCallback callback, object state)	Unknown
 	WindowsBase.dll!System.Windows.Threading.DispatcherOperation.Invoke()	Unknown
 	WindowsBase.dll!System.Windows.Threading.Dispatcher.ProcessQueue()	Unknown
 	WindowsBase.dll!System.Windows.Threading.Dispatcher.WndProcHook(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam, ref bool handled)	Unknown
 	WindowsBase.dll!MS.Win32.HwndWrapper.WndProc(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam, ref bool handled)	Unknown
 	WindowsBase.dll!MS.Win32.HwndSubclass.DispatcherCallbackOperation(object o)	Unknown
 	WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate callback, object args, int numArgs)	Unknown
 	WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.TryCatchWhen(object source, System.Delegate callback, object args, int numArgs, System.Delegate catchHandler)	Unknown
 	WindowsBase.dll!System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority priority, System.TimeSpan timeout, System.Delegate method, object args, int numArgs)	Unknown
 	WindowsBase.dll!MS.Win32.HwndSubclass.SubclassWndProc(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam)	Unknown
```

### PR DESCRIPTION
**Pull request by srivatsn**
Part of the change needed to fix #428. Shared projects will need to set the two new added properties to false to not get all the stuff we have by default. I'll create a PR in the devdiv repo for that.

@dotnet/project-system @davkean @jinujoseph 

Fixes #428, #450, #393, #394

### CODE REVIEW

File: src/Targets/Microsoft.Managed.DesignTime.targets [hunk 1]
```diff
@@ -76,11 +79,17 @@
       <Context>Project</Context>
     </PropertyPageSchema>
 
+    <PropertyPageSchema Include="$(ManagedXamlResourcesDirectory)ProjectDebugger.xaml">
```
Comment (333fred) [inline src/Targets/Microsoft.Managed.DesignTime.targets]: Mostly unrelated to this change, but does the ProjectDebugger.xaml file need to get added to `src\VsixV3\ProjectSystemPackage\Microsoft.VisualStudio.ProjectSystem.Managed.CommonFiles.swr`?

**Comment by 333fred** [comment]
:+1: module feedback.

File: src/Targets/Microsoft.Managed.DesignTime.targets (hunk 1 above)
Comment (srivatsn) [inline src/Targets/Microsoft.Managed.DesignTime.targets]: Added.

**Comment by jinujoseph** [comment]
:+1:

**Comment by mavasani** [comment]
👍

**Pull request merged**